frontend/packages/dev-console/src/utils/application-utils.ts

```diff
@@ -194,6 +211,100 @@ const updateItemAppConnectTo = (
   return k8sPatch(model, item, patch);
 };
 
+export const createServiceBinding = (
+  source: K8sResourceKind,
+  target: K8sResourceKind,
+): Promise<any> => {
+  if (!source || !target || source === target) {
+    return Promise.reject();
+  }
+
+  const targetName = _.get(target, 'metadata.name');
+  const sourceName = _.get(source, 'metadata.name');
+  const namespace = _.get(source, 'metadata.namespace');
+  const targetResourceGroup = _.split(_.get(target, 'metadata.ownerReferences[0].apiVersion'), '/');
+  const targetResourceKind = _.get(target, 'metadata.ownerReferences[0].kind');
+  const targetResourceRefName = _.get(target, 'metadata.ownerReferences[0].name');
+  const applicationSelectorLabels = getApplicationSelectorLabels(targetName);
+
+  const serviceBindingRequest = {
+    apiVersion: 'apps.openshift.io/v1alpha1',
+    kind: 'ServiceBindingRequest',
+    metadata: {
+      name: `${sourceName}-${targetName}-sbr`,
+      namespace,
+    },
+    spec: {
+      applicationSelector: {
+        matchLabels: { ...applicationSelectorLabels },
```

Comment:
Since this service binding is named specifically for binding the given source and target, shouldn't the selector be very specific to the target? I think the connects-to convention was intended to create a single service binding that can bind multiple apps to the service.

What happens when we have many service binding request resources that bind to the same service using the same selector?

If we intend to create a specific binding per source and target, it may be better to create a selector against the recommended labels of the source.

Thoughts @sbose78 ?
Is there only a selector? Or also a resource ref?

Reply:
You could use `resourceRef` too.

```
apiVersion: apps.openshift.io/v1alpha1
kind: ServiceBindingRequest
metadata:
  name: binding-request
  namespace: service-binding-demo
spec:
  applicationSelector:
    resourceRef: nodejs-app
    group: apps.openshift.io
    version: v1
    resource: deploymentconfigs
  backingServiceSelector:
    resourceRef: db-demo
    group: postgresql.baiju.dev
    version: v1alpha1
    kind: Database
```

Reply:
Awesome!
I think it makes more sense to create very targeted SBR in this case.